### PR TITLE
Fix to allow null and undefined if dictionary is expected

### PR DESCRIPTION
**Pull request by stefhak**
Relates to https://github.com/EricssonResearch/openwebrtc/issues/208. 
I note that there is some (for the Issue this PR is intended to address) garbage in the PR. Please review only the .js files
Also note that only if the expected type (or rather what the UA will translate to) is a dictionary, and only if the supplied argument is "undefined" it will be allowed without raising an error. It could be argued that any optional argument should be allowed to be undefined, and also that null should be allowed. But I think this is the right thing to do.

### CODE REVIEW

**Comment by adam-be** [comment]
Can we rename this issue and only fix the "allow null and undefined when dictionary is expected" issue? If we need to do something about optional arguments in general, we can do that in separate PRs.

File: bridge/client/domutils.js [hunk 1]
```diff
@@ -156,21 +156,22 @@ function getArgumentsError(argsTypeTemplate, numRequired, args) {
...
+function getTypeError(prefix, value, typeTemplate, optional) {
     var expectedTypes = typeTemplate.split(/\s*\|\s*/);
-    if (!canConvert(value, expectedTypes))
+    if (!canConvert(value, expectedTypes, optional))
         return prefix + " is of wrong type (expected " + expectedTypes.join(" or ") + ")";
     return null;
 }
 
-function canConvert(value, expectedTypes) {
+function canConvert(value, expectedTypes, optional) {
```
Comment (adam-be) [inline bridge/client/domutils.js]: I think we can wait with the "optional" argument and see if it's needed.

File: bridge/client/domutils.js [hunk 2]
```diff
@@ -184,11 +185,19 @@ function canConvert(value, expectedTypes) {
...
             if (constructor && value instanceof constructor)
                 return true;
         }
+        if (type == "undefined") {
+            if (expectedType == "dictionary") {
+                if (optional)
+                    return true;
+            }
+        }
         if (type == expectedType && expectedType == "function")
```
Comment (adam-be) [inline bridge/client/domutils.js]: To fix the "if dictionary is expected, then null and undefined are OK" issue, then we can check expectedType == "dictionary" and then return true if type is any of null, undefined or object.

Comment (stefhak) [inline bridge/client/domutils.js]: That would work for me. When/where will the actual content of the to-be dictionary be checked if for example setLocal is called with an undefined argument?

**Comment by adam-be** [comment]
Rebased and integrated.

**Pull request closed without merging**